### DISCUSSION TITLE
keep simulation running in transient (without changing anything)

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#18472] [Q&A Modules: General] [answered]
Hi guys,

For some reason, when I keep my simulation running without changing anything it just won't converge after t=3, not sure what's a problem.

The simulation works fine with the steady-state type

it's a basic flow simulation where left_BC is rho*g*h and right_BC is drainage

```
   [./water_grad_y1]
       type = FunctionDirichletBC
       variable = porepressure
       boundary = 'left'
       function = watergrady1
     [../]
    

          [./water_grad_right]
            type = PorousFlowPiecewiseLinearSink
          variable = porepressure
          boundary = 'right'
          pt_vals = '0 1e9' # x coordinates defining g
          multipliers = '0 1e9' # y coordinates defining g
          # PT_shift = 0  # BC pressure
          flux_function = 1e-2 # Variable C
          fluid_phase = 0
          [../]
```

this is the Executioner block

```
[Executioner]
  type = Transient
  start_time = 0
  end_time = 10
  dt = 1
  solve_type = NEWTON
  petsc_options = '-snes_converged_reason'
  petsc_options_iname = '-pc_type -pc_hypre_type -pc_hypre_boomeramg_strong_threshold -pc_hypre_boomeramg_agg_nl  -pc_hypre_boomeramg_agg_num_paths -pc_hypre_boomeramg_max_levels  -pc_hypre_boomeramg_coarsen_type   -pc_hypre_boomeramg_interp_type -pc_hypre_boomeramg_P_max -pc_hypre_boomeramg_truncfactor'
  petsc_options_value = 'hypre    boomeramg 0.5 4 5 25 HMIS  ext+i 2 0.3'
  line_search = bt
  nl_rel_tol = 5e-2
  l_max_its = 20
  nl_max_its = 1000
[]
```

![image](https://user-images.githubusercontent.com/79562439/127442767-441b978c-5ea7-4d93-b3bb-879b42cc5939.png)

my objective is to make the simulation a quasi-steady state, where I change the value of block_BC when the simulation reaches the steady state.

Kind regards,
Thanks

**Comment by hugary1995** [comment, marked as answer]
The default`nl_abs_tol = 1e-50` might be too small for you.

**Reply by WilkAndy** [reply]
I agree.  You have probably reached steady-state but MOOSE still wants to reduce the residual, but can't due to precision loss.  You might find something useful at https://mooseframework.inl.gov/modules/porous_flow/convergence.html or maybe https://mooseframework.inl.gov/modules/porous_flow/nonlinear_convergence_problems.html .

**Reply by Traiwit** [reply]
Hi guys, so I make `nl_abs_tol = 1e-5` and it works fine now (after a bit of fine-tuning)

So if I change anything after it reaches the steady-state, the residual will increase like to e-2 and this will be converged due to `nl_rel_tol`, but if it's at the steady-state, the system will be converged due to `nl_abs_tol`

not sure if this is the right way to do it, but it works for now.

Thank you all
Traiwit

**Reply by WilkAndy** [reply]
Yes, that is the correct way of setting the tolerances, according to me.  Choosing `nl_abs_tol` does usually require fine-tuning, since it depends on all your parameters (porosity, permeability, etc) and the size of your model.  Choosing `nl_rel_tol` is usually easier, since 1E-8 is usually sufficient to ensure reasonable solutions.

**Comment by GiudGiud** [comment]
nl_rel_tol = 5e-2 might be too high, you may be accepting totally unconverged solutions and putting the simulation in a bad spot

Same for l_max_its of 20, this is really small. 

Can you try a smaller timestep? This could help as well

**Reply by WilkAndy** [reply]
I agree with these comments too.
